tests/integration/test_users.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Discourse integration tests."""
+
+import logging
+
+import pytest
+import requests
+from juju.action import Action
+from juju.application import Application
+from juju.unit import Unit
+from pytest_operator.plugin import Model
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+async def test_create_user(
+    app: Application,
+):
+    """
+    arrange: A discourse application
+    act: Create a user
+    assert: User is created, and re-creating the same user should fail
+    """
+
+    await app.model.wait_for_idle(status="active")
+    discourse_unit: Unit = app.units[0]
+
+    email = "admin-user@test.internal"
+
+    action: Action = await discourse_unit.run_action("create-user", email=email)
+    await action.wait()
+    assert action.results["user"] == email
+
+    # Re-creating the same user should fail, as the user already exists
+    break_action: Action = await discourse_unit.run_action("create-user", email=email)
+    await break_action.wait()
+    assert break_action.status == "failed"
+
+
+@pytest.mark.asyncio
+async def test_promote_user(
+    app: Application,
+    discourse_address: str,
+    model: Model,
+    requests_timeout: float,
+):
+    """
+    arrange: A discourse application
+    act: Promote a user to admin
+    assert: User cannot access the admin API before being promoted
+    """
+
+    def test_discourse_srv_status_ok():
+        response = requests.get(f"{discourse_address}/srv/status", timeout=requests_timeout)
+        assert response.status_code == 200
+
+    # The charm should be active when starting this test
+    await model.wait_for_idle(status="active")
+    test_discourse_srv_status_ok()
```

Comment:
Not sure why this is needed. 

In the previous test this check is not there. If this test needs it, the previous one also needs it. I would say that none of them needs it. If the unit is active, we should be able to run the test.

Reply:
i don't think either needs it, it just seemed as though it had fixed some saml issues yesterday but this was not the case